### PR TITLE
Add support for the `images={}` attribute.

### PR DESCRIPTION
**Pull request by mattmoor**
When the `images={}` attribute is specified resolution will publish the collection of images, and substitute the exact digest of the published image as part of resolution (avoids race condition).

### CODE REVIEW

**Comment by mattmoor** [comment]
@dlorenc It is notable that this won't work with `minikube` unless the user's roundtripping through a real registry.  

(thinking out loud)
That said, we can probably put together an alternate implementation of `Publish()` that supports `docker load`ing the image into `minikube`'s docker daemon.  If we put this in a little library in `rules_docker` then we could also use it to implement [this](https://github.com/bazelbuild/rules_docker/issues/72), which is a nice two-for-one.

WDYT?

File: k8s/object.bzl [hunk 1]
```diff
@@ -23,25 +33,70 @@ def _impl(ctx):
...
+        ctx.attr.image_targets, ctx.attr.image_target_strings)
+    image_files_dict = _string_to_label(
+        ctx.files.image_targets, ctx.attr.image_target_strings)
+
+    # Walk the collection of images passed and for each key/value pair
+    # collect the parts to pass to the resolver as --image_spec arguments.
+    # Each images entry results in a single --image_spec argument.
+    # As part of this walk, we also collect all of the image's input files
+    # to include as runfiles, so they are accessible to be pushed.
+    index = 0
```
Comment (dlorenc) [inline k8s/object.bzl]: Is index used anywhere?

Comment (mattmoor) [reply k8s/object.bzl]: Yeah, where I increment it :)  Removed.

**Comment by mattmoor** [comment]
please test this

**Comment by mattmoor** [comment]
test this please

**Comment by rohansingh** [comment]
> That said, we can probably put together an alternate implementation of Publish() that supports docker loading the image into minikube's docker daemon.

@mattmoor I've put together some rules and a macro for a repo that basically does this. Notes for anyone that wants to take a stab on a real implementation:

1. The difficult part was deciding how to tag the images locally, since locally-loaded Docker images can't be referenced by digest. I ended up just tagging the images with their digest value.

2. I didn't want to tackle [rewriting the incremental loader](https://github.com/bazelbuild/rules_docker/issues/72) from `rules_docker` in Python. Instead, I just created a separate rule that wraps `incremental_load` from `@io_bazel_rules_docker//container:layer_tools.bzl` to load the image, and then wraps `k8s_object` to do Kube stuff.

3. Because of the above, I didn't change `Publish()` to load the image into the Docker daemon. Instead, the resolver just accepts a `--no_upload` flag. When passed, it does not upload the image. Instead, it resolves the image name using the tag strategy described above.

Obviously this is a bit of a hacky solution, but it seems to work quite reliably. Having the ability to deploy manifests locally like this is really handy.

**Comment by mattmoor** [comment]
FWIW I solved this in `ko` by simply tagging the image with the digest, see: https://github.com/google/go-containerregistry/blob/master/cmd/ko/README.md

**Comment by mattmoor** [comment]
Note: my bias towards not using `:latest` was that it doesn't work well with K8s default pull policy.

**Pull request merged**